tests/foreman/cli/test_discoveryrule.py

```diff
@@ -567,46 +573,29 @@ def test_negative_update_priority(self):
 
         :CaseImportance: Medium
         """
-        rule = self._make_discoveryrule()
+        rule = discoveryrule_factory()
         priority = gen_string('alpha')
-        with self.assertRaises(CLIReturnCodeError):
-            DiscoveryRule.update({'id': rule['id'], 'priority': priority})
+        with pytest.raises(CLIReturnCodeError):
+            DiscoveryRule.update({'id': rule.id, 'priority': priority})
 
 
-class DiscoveryRuleRoleTestCase(CLITestCase):
+class TestDiscoveryRuleRole:
```

Comment:
Currently debugging some test failures in this class due to permissions of the created user.